patterns/strategy.md

```diff
@@ -0,0 +1,171 @@
+# Strategy (aka Policy)
+
+## Description
+
+The [Strategy design pattern](https://en.wikipedia.org/wiki/Strategy_pattern)
+is a technique that enables separation of concerns in object-oriented software design.
```

Comment:
Is this really related to object-oriented software design? I can't see how it relates.

Reply:
@pickfire We may discuss here on what OOD is for days and stil can't come to a consensus, can we?. Rust book ([chapter 17]( https://doc.rust-lang.org/book/ch17-01-what-is-oo.html)) writes:

> There is no consensus in the programming community about what features a language must have to be considered object oriented.
Nevertheless, in my opinion, if you are dealing with DATA + BEHAVIOUR then we traditionally call them "OBJECTS", and if we think in terms of "OBJECTS" then we simply call it object oriented.  

As for design patterns, they are simply a set of principles (or whatever you call them) to follow to make software design clear, easy to refactor, readable, etc.
The main point of this section is the Strategy pattern and not whether or not it relates to OOD or what OOD is.

Reply:
Okay, I guess I am fine with it after you explained, but I rarely see the docs mentioning about objects which make me a bit confused.

>  The main point of this section is the Strategy pattern and not whether or not it relates to OOD or what OOD is.

Yes, that is why I mention this. Because could be done in non OOD as well, since it could be separated from file level.